### ISSUE TITLE
Google login not working in local installation

### ISSUE THREAD

**Issue by alphalambda** [google/codeworld#123]
I have a local installation of codeworld that I am testing, but I cannot get to have a working Google login.
I followed the instructions and created an OAuth 2.0 client ID in the Google Developers Console. I also enabled the Gmail and Google+ APIs there. Then I copied the ClientID from the Google Developers Console into a file, which I placed in web/clientId.txt
However, when I click login in codeworld, I get a new browser window with this url:
https://accounts.google.com/o/oauth2/auth?client_id= ... (rest omitted)

The text of the page says:
401. That's an error.
Error: invalid_client
The OAuth client was not found.
Request Details
That's all we know.

I have no clue what is wrong, whether it is a bug in the code or in the documentation, because maybe there is something else I should be doing, but I am not a web developer, so I don't know what.

**Comment by alphalambda** [comment]
I forgot to mention that I also tried creating a Browser Key and also a Server Key, but the error message from Google was the same.
The Google Developers Console is very confusing, so maybe I need a different key or a different API enabled?

**Comment by cdsmith** [comment]
I'm not sure where you're creating these things, but here's what I did:

1. Create a project at http://console.cloud.google.com
2. Go to https://console.cloud.google.com/apis/credentials?project=your_project
3. Click "Create credentials", and create an oauth client ID.
4. Choose Web Application as the client type.
5. Include the base URL of your local server, as an authorized JavaScript origin.

The client ID ends with ".apps.googleusercontent.com"  It should be the only contents of that file.  There shouldn't even be a trailing newline!  I just verified that adding a trailing newline gives the same error you posted.  I'll fix that in a minute.
